File: problems/problems_874/solution.py
import solution
from typing import *


class Solution(solution.Solution):
    def solve(self, test_input=None):
        return self.robotSim(*test_input)

    def robotSim(self, commands: List[int], obstacles: List[List[int]]) -> int:
        obstacles_ = set(map(tuple, obstacles))
        dirs = [(0, 1), (1, 0), (0, -1), (-1, 0)]
        x = y = d = ans = 0

        for c in commands:
            if c < 0:
                d = (d + (1 if c == -1 else 3)) % 4
            else:
                dx, dy = dirs[d]
                for _ in range(c):
                    if (x + dx, y + dy) in obstacles_:
                        break
                    x, y = x + dx, y + dy
                ans = max(ans, x * x + y * y)
        return ans

File: problems/problems_874/Solution.java
package problems.problems_874;

import com.alibaba.fastjson.JSON;
import java.util.*;
import qubhjava.BaseSolution;


public class Solution extends BaseSolution {
    public int robotSim(int[] commands, int[][] obstacles) {
        
    }

    @Override
    public Object solve(String[] inputJsonValues) {
        int[] commands = jsonArrayToIntArray(inputJsonValues[0]);
		int[][] obstacles = jsonArrayToInt2DArray(inputJsonValues[1]);
        return JSON.toJSON(robotSim(commands, obstacles));
    }
}
